File: everyDayGo/1732.找到最高海拔.go
/*
 * @lc app=leetcode.cn id=1732 lang=golang
 *
 * [1732] 找到最高海拔
 */

// @lc code=start
func largestAltitude(gain []int) (res int) {
	res, sum := 0, 0
	for _, i := range gain {
		sum += i
		res = max(res, sum)
	}
	return
}
func max(a, b int) int {
	if b > a {
		return b
	}
	return a
}

// @lc code=end

File: everyDay/1732.找到最高海拔.cpp
/*
 * @lc app=leetcode.cn id=1732 lang=cpp
 *
 * [1732] 找到最高海拔
 */

// @lc code=start
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int res = 0, sum = 0;
        for (int i = 0; i < gain.size(); i++ ) {
            sum += gain[i];
            res = max(sum, res);
        }
        return res;
    }
};
// @lc code=end

